treescript/src/treescript/gecko/android_l10n.py

```diff
@@ -0,0 +1,206 @@
+        files = paths.ProjectFiles(locale, [project_config])
+
+        for l10n_file, reference_file, _, _ in files:
+            # Ignore missing files for locale
+            if not os.path.exists(l10n_file):
+                continue
+            # Ignore if reference file does not exist
+            if not os.path.exists(reference_file):
+                continue
+
+            l10n_files.append(
+                {
+                    "abs_path": l10n_file,
+                    "rel_path": os.path.relpath(l10n_file, reldir),
+                }
+            )
+
+    return l10n_files
+
+
+# copy_android_l10n_files {{{1
+def copy_android_l10n_files(l10n_files, src_repo_path, dest_repo_path):
+    """Copy localized files in code repository"""
+
+    log.info(f"Files to copy: {len(l10n_files)}.")
+    for l10n_file in l10n_files:
+        if src_repo_path:
+            src_file = os.path.join(src_repo_path, l10n_file["rel_path"])
+        else:
+            src_file = l10n_file["abs_path"]
+        dest_file = os.path.join(dest_repo_path, l10n_file["rel_path"])
+        log.info(f"  {src_file} -> {dest_file}")
+        # Make sure that the folder exists, then copy file as is
+        os.makedirs(os.path.dirname(dest_file), exist_ok=True)
+        shutil.copy2(src_file, dest_file)
+
+
+# build_commit_message {{{1
+def build_commit_message(description, dontbuild=False, ignore_closed_tree=False):
+    """Build a commit message for android-l10n import.
+
+    Args:
+        dontbuild (bool, optional): whether to add ``DONTBUILD`` to the
+            comment. Defaults to ``False``
+        ignore_closed_tree (bool, optional): whether to add ``CLOSED TREE``
+            to the comment. Defaults to ``False``.
+
+    Returns:
+        str: the commit message
+
+    """
+    approval_str = "r=release a=android_l10n-import"
+    if dontbuild:
+        approval_str += DONTBUILD_MSG
+    if ignore_closed_tree:
+        approval_str += CLOSED_TREE_MSG
+    message = f"no bug - {description} {approval_str}\n\n"
+    return message
+
+
+# android_l10n_action {{{1
+async def android_l10n_action(config, task, repo_path, description, is_import):
+    """Perform a android_l10n string import or merge action.
+
+    Args:
+        config (dict): the running config
+        task (dict): the running task
+        repo_path (str): the source directory
+        description (str): commit message description
+        is_import (bool): True if import action
+
+    Returns:
+        int: non-zero if there are any changes.
+
+    """
+    ignore_closed_tree = get_ignore_closed_tree(task)
+    if not ignore_closed_tree:
+        if not await check_treestatus(config, task):
+            log.info("Treestatus is closed; skipping android-l10n import.")
+            return 0
+
+    dontbuild = get_dontbuild(task)
+
+    if is_import:
+        task_info = get_android_l10n_import_info(task)
+    else:
+        task_info = get_android_l10n_merge_info(task)
```

Comment:
Can we avoid conditionals here?  E.g. pass the `get_android_l10n_*_info` as an argument instead of the `is_import` boolean, and similar below, extract the different paths out of this function.

Reply:
We can but it gets tricky and introduces a lot of parameters. I'll give it a try but I'm also considering separating import and sync entirely (removing android_l10n_action).